opendatahub-io/opendatahub-operator
# Initialize Internal APIs for components

## Description
This PR includes following changes. Some of the items are yet to be completed:
- [x] Define Internal API structs for components
- [x] Implement one component controller (Dashboard)
- [x] Move away from ComponentInterface
- [x] Update and add Tests
- [x] Update an add statuses
- [x] Update the DataScienceCluster controller

## How Has This Been Tested?

CatalogSource: quay.io/vhire/opendatahub-operator-catalog:v2.19.4 

1. Install operator by deploying above catalogsource
2.  Create DSCI
3. Create DSC (only enable Dashboard)
4. Verify `Dashboard` CR is created
5. Verify Dashboard resources are deployed




## Screenshot or short clip


## Merge criteria
